src/config/cloudtrail.json

```diff
@@ -0,0 +1,1006 @@
+        },
+        {
+          "key": "aws.resource.id",
+          "pattern": "if( log_content_parsed.eventName == 'DeleteDBCluster', &log_content_parsed.requestParameters.dBClusterIdentifier, &EmptyValue, @ )"
+        },
+        {
+          "key": "dt.source_entity",
+          "pattern": "if( log_content_parsed.eventName == 'DeleteDBInstance', &dt_meid_rds_v2( \"aws.arn\"), &EmptyValue, @ )"
+        },
+        {
+          "key": "aws.arn",
+          "priority": 10,
+          "pattern": "if( log_content_parsed.eventName == 'DeleteDBInstance', &format_arn('arn:{}:rds:{}:{}:db:{}', [partition, region, account_id, log_content_parsed.requestParameters.dBInstanceIdentifier]), &EmptyValue, @ )"
+        },
+        {
+          "key": "aws.resource.id",
+          "pattern": "if( log_content_parsed.eventName == 'DeleteDBInstance', &log_content_parsed.requestParameters.dBInstanceIdentifier, &EmptyValue, @ )"
+        },
+        {
+          "key": "dt.source_entity",
+          "pattern": "if( log_content_parsed.eventName == 'DeleteDBInstanceAutomatedBackup', &dt_meid_rds_v2( \"aws.arn\"), &EmptyValue, @ )"
+        },
+        {
+          "key": "aws.arn",
+          "priority": 10,
+          "pattern": "if( log_content_parsed.eventName == 'DeleteDBInstanceAutomatedBackup', &format_arn('arn:{}:rds:{}:{}:db:{}', [partition, region, account_id, log_content_parsed.requestParameters.DbiResourceId]), &EmptyValue, @ )"
+        },
+        {
+          "key": "aws.resource.id",
+          "pattern": "if( log_content_parsed.eventName == 'DeleteDBInstanceAutomatedBackup', &log_content_parsed.requestParameters.DbiResourceId, &EmptyValue, @ )"
+        },
+
+        {
+          "key": "dt.source_entity",
+          "pattern": "if( contains(log_content_parsed.eventName, 'DeleteEventSubscription') && log_content_parsed.requestParameters.sourceType == 'db-cluster', &dt_meid_supporting_service_v2( 'Aurora', \"aws.arn\"), &EmptyValue, @ )"
+        },
+        {
+          "key": "aws.arn",
+          "priority": 10,
+          "pattern": "if( contains(log_content_parsed.eventName, 'DeleteEventSubscription') && log_content_parsed.requestParameters.sourceType == 'db-cluster', &format_arn('arn:{}:rds:{}:{}:cluster:{}', [partition, region, account_id, log_content_parsed.responseElements.sourceIdsList[0] ]), &EmptyValue, @ )"
+        },
+        {
+          "key": "aws.resource.id",
+          "pattern": "if( contains(log_content_parsed.eventName, 'DeleteEventSubscription') && log_content_parsed.requestParameters.sourceType == 'db-cluster', &log_content_parsed.responseElements.sourceIdsList[0], &EmptyValue, @ )"
+        },
+        {
+          "key": "dt.source_entity",
+          "pattern": "if( contains(log_content_parsed.eventName, 'DeleteEventSubscription') && log_content_parsed.requestParameters.sourceType == 'db-instance', &dt_meid_rds_v2( \"aws.arn\"), &EmptyValue, @ )"
```

Comment:
Yeah, same case here, lots of contains check that might be unnecessary in many cases.
Actually this might be optimised further if our switch makes a dictionary with keys of possible values ("DeleteEventSubscription") and then reference specific dictionary entry and evaluate this. 
I am not sure how big perf impact is but this metadata thing is run for each single entry so I think this is worth investigating.

Reply:
I think this is actually a classic way for switches, e.g. compiled C does this, where it does not evaluate each condition but jumps straight to correct one with O(1). There are some assumptions to be held though

Reply:
And if we raise this switch-dictionary to schema (new field: "conditionalPattern" with "sourceField" and a map value->expression) then this dictionary is built only once when loading configs

Reply:
Eeee, this contains should be gone, it's a left-over from an attempt to have reduce the number of ifs by using *.